raiden/storage/serialize.py

```diff
@@ -25,17 +24,6 @@ class RaidenJSONDecoder(json.JSONDecoder):
     its state.
     """
 
-    # Maintain a global cache instance which lives across different decoder instances
-    # The reason we need this ref cache is:
-    # State Objects are currently mutable. Different State Machines keep pointers of the
-    # same object such as channel states, balance proofs .. etc.
-    # Therefore, we rely on the fact that any "shared" object changes should be seen by
```

Comment:
Hmmm but doesn't completely removing the reference cache cause the exact problem for which we introduced it? It's described in this comment here.

Reply:
I removed the cycles too

Reply:
aha I see. So by changing `_to_channels` to `_to_channelidentifiers` and `_to_tokens` to `_to_tokenidentifiers` you are also eliminating the reference cycles for which the reference cache was introduced. That's awesome.